From: Robert Luberda <robert@debian.org>
Date: Sat, 2 Jul 2016 22:23:20 +0200
Subject: Switch to _Static_assert

Use _Static_assert for compile-time assertion to fix
build failures with gcc-6 (closes: #811707)
---
 acc/acc_defs.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/acc/acc_defs.h b/acc/acc_defs.h
index 866b7bd..5ee3761 100644
--- a/acc/acc_defs.h
+++ b/acc/acc_defs.h
@@ -87,6 +87,9 @@
 
 /* This can be put into a header file but may get ignored by some compilers. */
 #if !defined(ACC_COMPILE_TIME_ASSERT_HEADER)
+# define ACC_COMPILE_TIME_ASSERT_HEADER(e) _Static_assert(e, #e);
+#endif
+#if !defined(ACC_COMPILE_TIME_ASSERT_HEADER)
 #  if (ACC_CC_AZTECC || ACC_CC_ZORTECHC)
 #    define ACC_COMPILE_TIME_ASSERT_HEADER(e)  extern int __acc_cta[1-!(e)];
 #  elif (ACC_CC_DMC || ACC_CC_SYMANTECC)
@@ -100,6 +103,9 @@
 
 /* This must appear within a function body. */
 #if !defined(ACC_COMPILE_TIME_ASSERT)
+# define ACC_COMPILE_TIME_ASSERT(e) _Static_assert(e, #e);
+#endif
+#if !defined(ACC_COMPILE_TIME_ASSERT)
 #  if (ACC_CC_AZTECC)
 #    define ACC_COMPILE_TIME_ASSERT(e)  {typedef int __acc_cta_t[1-!(e)];}
 #  elif (ACC_CC_DMC || ACC_CC_PACIFICC || ACC_CC_SYMANTECC || ACC_CC_ZORTECHC)
